compiler/rustc_type_ir/src/sty.rs

```diff
@@ -499,33 +499,53 @@ impl<I: Interner> hash::Hash for TyKind<I> {
 impl<I: Interner> fmt::Debug for TyKind<I> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
-            Bool => f.write_str("Bool"),
```

Comment:
Why this change? If you want a rust like syntax you can print `Ty` instead of `TyKind`. On multiple occasions I have printed `ty.kind()` precisely because I wanted to know the internal representation with all details rather than a nicely formatted representation.

Reply:
I don't oppose having a nice syntax as long as it's like... "injective" in the mathematical sense. Like, pretty printing does things like rebinding late bound vars and eliding lifetimes and leaving out defaulted adt substs (I think), but my understanding is that the aim of this PR is to continue conveying the same amount of information, but in a bit more native format. Future changes on top to actually make `Ty: Debug` call to `TyKind: Debug` should avoid going thru any of those misleading output changes for the recursive calls to debug.

I guess I'm also not opposed to keeping the output a bit more "`derive(Debug)`-like" (like using parentheses as delimiters for info).

Reply:
You got me wrong. I mean that I prefer if `Debug for TyKind` stays the same, and instead `Debug for Ty` uses pretty printing and if you want pretty printing you can format `Ty` instead of `TyKind`.

Reply:
I don't see any win for printing as `Bool` over `bool` otoh `bool` is more like the syntax everybody knows. debug printing `ty.kind()` is reasonable rn since `Ty` doesn't actually have a Debug impl. `TyKind` just does not work as a debug impl, its far too verbose to reasonably use to actually debug anything.

The only important thing for a `Debug` impl here is that it doesnt lose any important information (which this PR doesnt cause us to do). It's already too late for `TyKind::Debug` to be "just how we represent", infer vars have a debug impl that is more concise and using the notation that people use to talk about infer vars in technical convos. It makes more sense for the debug impl for how we represent types to be useful for people debugging the type system than for people who dont know how `TyKind` works.

We actually cant `{:?}` `Ty` right now either, that's just a wrapper for `with_no_trimmed_paths` around the display impl (ofc eventually we should change this to defer to `TyKind: Debug`). The goal is not for `TyKind: Debug` to be "rust syntax", the goal is for it to be an output suitable for debugging a `TyKind`. Right now that is not possible.